subprojects/docs/src/docs/release/notes.md

```diff
@@ -148,6 +148,17 @@ This avoids a 2nd request for the JAR file when the POM is missing, making depen
 
 You can also use this to programmatically opt-in to using the new Gradle metadata format for one repository using `metadataSources { gradleMetadata() }`.
 
+### Specifying reasons for dependency declaration and resolution rules
+
+In complex builds, dependency resolution results can be hard to interpret. Sometimes the reason why a [dependency declaration](userguide/customizing_dependency_resolution_behavior.html) or a [rule](userguide/inspecting_dependencies.html#sec:dependency_declaration_reasons) was added to the build script can get lost. To improve on this situation, we extended all the corresponding APIs with the capability to define a _reason_ for each declaration or rule. These reasons are shown in dependency insight reports and error messages if the corresponding declaration or rule influenced the resolution result. In the future, they will also be shown in build scans.
```

Comment:
I am not quite sure how the reason can get lost. Maybe users don't know how to look for it via the dependencyInsight report or a build scan? We should be more specific here.

Reply:
With "reason can get lost" I was not referring to the reason concept in Gradle but just the reason in the head of the author when editing the build script. I reworded it.

Reply:
That wasn't necessarily clear to me. Thanks!